returnn/tf/layers/basic.py

```diff
@@ -1341,6 +1341,17 @@ def __init__(self, position, axis, **kwargs):
     # (BatchAxes.., InputAxesBeforeGatherAxis, PositionAxes.., InputAxesAfterGatherAxis..)
     self.output.placeholder = tf.gather(params=params, indices=indices, axis=gather_axis, batch_dims=batch_dims)
 
+    if input_data.dim_tags[old_gather_axis].is_batch_dim():
+      for axis in self.output.size_placeholder:
+        new_size = tf.gather(params=self.output.size_placeholder[axis], indices=position_data.placeholder)
+        from ..util.data import Dim
+        Dim(
+          kind=Dim.Types.Spatial, description="%s_gather_axis" % self.name,
+          dyn_size=new_size, batch=self.output.batch,
```

Comment:
You miss `dyn_size_ext` here.

Reply:
You should not set `dyn_size` in case it is non-standard. Set `dyn_size_ext` instead.